browser/ui/webui/brave_web_ui_controller_factory.cc

```diff
@@ -187,35 +195,42 @@ WebUIFactoryFunction GetWebUIFactoryFunction(WebUI* web_ui, const GURL& url) {
   return nullptr;
 }
 
-#if BUILDFLAG(IS_ANDROID)
 bool ShouldBlockRewardsWebUI(content::BrowserContext* browser_context,
                              const GURL& url) {
   if (url.host_piece() != kRewardsPageHost &&
+#if !BUILDFLAG(IS_ANDROID)
+      url.host_piece() != kTipHost &&
+      url.host_piece() != kBraveRewardsPanelHost &&
+#endif  // !BUILDFLAG(IS_ANDROID)
       url.host_piece() != kRewardsInternalsHost) {
     return false;
   }
-  if (!base::FeatureList::IsEnabled(brave_rewards::features::kBraveRewards)) {
-    return true;
-  }
+
   Profile* profile = Profile::FromBrowserContext(browser_context);
-  if (profile && profile->GetPrefs() &&
-      profile->GetPrefs()->GetBoolean(kSafetynetCheckFailed)) {
-    return true;
+  if (profile) {
+    if (!brave_rewards::IsSupportedForProfile(profile)) {
+      return true;
+    }
+    auto* prefs = profile->GetPrefs();
+    if (prefs) {
```

Comment:
We could move these in the `IS_ANDROID` section.

Reply:
Great catch! Done in https://github.com/brave/brave-core/pull/14928/commits/8bfd36f672940008123eda01889f44d90d14e7c2